core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -248,6 +251,16 @@ private[spark] class BlockManager(
     logInfo(s"Initialized BlockManager: $blockManagerId")
   }
 
+  def shuffleMetricsSource: Source = {
+    import BlockManager._
+
+    if (externalShuffleServiceEnabled) {
+      new ShuffleMetricsSource("ExternalShuffle", shuffleClient.shuffleMetrics())
+    } else {
+      new ShuffleMetricsSource("NettyBlockTransfer", shuffleClient.shuffleMetrics())
```

Comment:
do you think we really need to distinguish these two cases?  whether or not you have the external shuffle service, this memory is still owned by the executor JVM (its really only external on the remote end).

Reply:
For the external shuffle, we only have Transport client in the executor side, while for `NettyBlockTransfer` each executor will both server as transport client as well as server. So from my thought I explicitly distinguish those two cases.

Reply:
ok, I guess I haven't seen enough setups to know how users subscribe to these and whether they'd actually want that distinction.  (especially since it seems like some of the distinction probably shouldn't be there, eg. the openBlockLatency metric.)  But I don't think there is any clear right answer here, if you think this is the best naming, that is fine with me.